### PR TITLE
Local/remote dev setup

### PR DESCRIPTION
**Pull request by ziyangczi**
Documentation taken from michael's branch
adjusted the workflow to the team discussion:
- continue with current deployment flow and maintain staging - main branch, prod - 2 weeks behind staging
- add remote dev deployment with prefix dev-

### CODE REVIEW

File: .github/workflows/push-rdev.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+name: Push Remote Dev
```
Comment (ziyangczi) [inline .github/workflows/push-rdev.yml:1]: This workflow enables automatic stack creation from branch dev-*, which is only available to CZI employees since we don't give write permission to others in repo

File: scripts/happy-deploy.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
```
Comment (codemonkey800) [inline scripts/happy-deploy.py:1]: what is this script for? is it for starting a deployment workflow on GitHub?

it would be nice to have some documentation about this, but it can be in another PR if you're working on docs in another branch 😄

File: .github/workflows/push-rdev.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+on:
+  push:
+    branches:
+      - 'dev-*'
```
Comment (codemonkey800) [inline .github/workflows/push-rdev.yml:6]: it would be nice if we could have some kind of cleanup when the branch gets deleted. There is a way to listen for [delete events](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#delete).

**Review by codemonkey800 — APPROVED** [review]
LGTM 🍜 

I left a few comments but they don't have to be addressed in this PR. I'm just excited to get the dev branches 😄

File: .github/workflows/push-rdev.yml (hunk 3 above)
Comment (ziyangczi) [reply .github/workflows/push-rdev.yml:6]: added but untested, will see how it works

File: scripts/happy-deploy.py (hunk 2 above)
Comment (ziyangczi) [reply scripts/happy-deploy.py:1]: this is what infra is pushing for future deployment flows, just going to leave it in here until we want to use this

Comment (ziyangczi) [reply scripts/happy-deploy.py:1]: just to be clear it is not being referenced right now
